### PR TITLE
Adding Avatar docs

### PR DESCRIPTION
**Pull request by ktabors**
Questions
- Based on Spectrum calling Avatar "Navigation", I put it in that category.
- Is Avatar decorative like an image or will it be in a taggroup or wrapped by a link? Should an example show that?
- Should we use behance of imgur?

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [X] Filled out test instructions.
- [X] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Please read the docs.

## 🧢 Your Project:
RSP

### CODE REVIEW

File: packages/@react-spectrum/avatar/docs/Avatar.mdx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+<HeaderInfo
+  packageData={packageData}
+  componentNames={['Avatar']}
+  sourceData={[{type: 'Spectrum', url: 'https://spectrum.adobe.com/page/avatar/'}]} />
+
+## Example
+
+```tsx example
+<Avatar src="https://mir-s3-cdn-cf.behance.net/project_modules/disp/690bc6105945313.5f84bfc9de488.png" alt="default Adobe avatar" />
```
Comment (reidbarber) [inline packages/@react-spectrum/avatar/docs/Avatar.mdx:40]: Not a huge deal, but maybe we could use a shorter url. We could link this via imgur.com, like we have on the Image docs. It also seems like then the image would be less likely to disappear?

Comment (ktabors) [reply packages/@react-spectrum/avatar/docs/Avatar.mdx:40]: right, I meant to ask if the URL should be imgur or behance.

File: packages/@react-spectrum/avatar/docs/Avatar.mdx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+## Content
+
+The Avatar component is a thumbnail representation of an entity, such as a user or an organization.
+
+**Note: Avatars must have an `alt` attribute.** If the avatar is considered decorative and should not be announced by screen readers, then set `alt=""` to suppress the warning.
```
[the remark this answers is not in the document]
Comment (ktabors) [reply packages/@react-spectrum/avatar/docs/Avatar.mdx:47]: This language was copied from Image. I looked at the prop table and noticed that image's `alt` prop doesn't have a default while Avatar's default is `null`. It seems like these should be the same.

[the remark this answers is not in the document]
Comment (reidbarber) [reply packages/@react-spectrum/avatar/docs/Avatar.mdx:47]: Sounds good. We can fix in a separate non-docs PR.

File: packages/@react-spectrum/avatar/docs/Avatar.mdx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import {HeaderInfo, PropTable, PageDescription} from '@react-spectrum/docs';
+import packageData from '@react-spectrum/avatar/package.json';
+
+```jsx import
+import {Avatar} from '@react-spectrum/avatar';
+import {Flex} from '@react-spectrum/layout';
+```
+
+---
+category: Navigation
```
[the remark this answers is not in the document]
Comment (matthewdeutsch) [reply packages/@react-spectrum/avatar/docs/Avatar.mdx]: agreed, Content feels like a better fit

**Review by snowystinger — APPROVED** [review]

**Pull request merged**